libs/SalesforceSDK/src/com/salesforce/androidsdk/ui/LoginActivity.kt

```diff
@@ -675,144 +693,193 @@ open class LoginActivity : AppCompatActivity(), OAuthWebviewHelperEvents {
     // region QR Code Login Via UI Bridge API Public Implementation
 
     /**
-     * Automatically log in with a UI Bridge API login QR code.
-     * @param loginQrCodeContent The login QR code content.  This should be
-     * either a URL or URL query containing the UI Bridge API JSON parameter.
-     * The UI Bridge API JSON parameter should contain URL-encoded JSON with two
-     * values:
-     * - frontdoor_bridge_url
-     * - pkce_code_verifier
+     * Automatically log in with a UI Bridge API front door bridge URL and PKCE code verifier.
+     *
+     * This method is the intended entry point to Salesforce Mobile SDK when using the Salesforce
+     * Identity API UI Bridge front door URL.  Usable, default implementations of methods are
+     * provided for parsing the UI Bridge parameters from the reference JSON and log in URLs used
+     * by the reference QR Code Log In implementation.  However, the URL and JSON structure in the
+     * reference implementation is not required.  An app may use a custom structure so long as this
+     * entry point is used to log in with the front door URL and optional PKCE code verifier.
      *
-     * If pkce_code_verifier is not specified then the user agent flow is used
-     * @return Boolean true if a log in attempt is possible using the provided QR
-     * code content, false otherwise
-     */
-    fun loginFromQrCode(
-        loginQrCodeContent: String?
-    ) = uiBridgeApiParametersFromLoginQrCodeContent(
-        loginQrCodeContent
-    )?.let { uiBridgeApiParameters ->
-        loginWithFrontdoorBridgeUrl(
-            uiBridgeApiParameters.frontdoorBridgeUrl,
-            uiBridgeApiParameters.pkceCodeVerifier
-        )
-        true
-    } ?: false
-
-    /**
-     * Automatically log in with a UI Bridge API front door bridge URL and PKCE
-     * code verifier.
      * @param frontdoorBridgeUrl The UI Bridge API front door bridge URL
-     * @param pkceCodeVerifier The PKCE code verifier
+     * @param pkceCodeVerifier The optional PKCE code verifier, which is not required for User Agent
+     * Authorization Flow but is required for Web Server Authorization Flow
      */
     @Suppress("MemberVisibilityCanBePrivate")
     fun loginWithFrontdoorBridgeUrl(
```

Comment:
This is our universal entry point to MSDK's front door bridge URL login logic, even if the app chooses not to use our reference implementation's APEX, Visualforce Page and the associated login QR code URL/JSON format.